system/system_windows.go

```diff
@@ -72,67 +73,120 @@ func removeUninstallEntry(name string) error {
 	return registry.DeleteKey(registry.LOCAL_MACHINE, reg)
 }
 
-func uninstallString(publisher, installSource, programName, extension string) string {
-	productroot := `SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`
-	k, err := registry.OpenKey(registry.LOCAL_MACHINE, productroot, registry.ENUMERATE_SUB_KEYS)
-	if err != nil {
-		return ""
+func AppAssociation(publisher, installSource, programName, extension string) (string, string) {
+
+	var productroots = []string{
+		`SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`,
+		`SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\`,
 	}
-	reg, _ := k.ReadSubKeyNames(-1)
-	defer k.Close()
-	for _, v := range reg {
-		q, err := registry.OpenKey(registry.LOCAL_MACHINE, fmt.Sprintf("%s%s", productroot, v), registry.ALL_ACCESS)
-		defer q.Close()
+	for _, productroot := range productroots {
+		k, err := registry.OpenKey(registry.LOCAL_MACHINE, productroot, registry.ENUMERATE_SUB_KEYS)
 		if err != nil {
-			continue
+			return "", ""
 		}
-		switch extension {
-		case "msi":
-			a, _, err := q.GetStringValue("InstallSource")
+		reg, _ := k.ReadSubKeyNames(-1)
```

Comment:
It's a bit weird that this proceeds to use the reg value even if an error was returned, but this seems to just be what ReadSubKeyNames expects one to do?